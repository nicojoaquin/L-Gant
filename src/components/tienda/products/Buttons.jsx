import React, {useState, useRef} from 'react'
import PropTypes from 'prop-types'

const Buttons = ({buttonText, setItems, items}) => {
  
  const [disabled, setDisabled] = useState(false)
  const addButton = useRef();

  const newProduct = {
    id: null,
    name: "Zapatos marrones",
    price: 16600, 
    category: "zapatos",
    img: "/assets/products/zapato_marron.jpg"
  }

  const addProduct = () => {
    newProduct.id = items.length + 1
    setItems([newProduct, ...items])
    setDisabled(true)
  }

  return (

    <div className= "buttons">

        <button 
          className = 'addButton' 
          ref = {addButton} 
          disabled = {disabled}
          onClick = { addProduct }
          >{buttonText}          
        </button>

        <button
          className = "changeColor" 
          disabled = {disabled}
          onClick = { () => {
            addButton.current.style.background = "green"
          }}
          >Cambiar color
        </button>

    </div>

  )

}

Buttons.propTypes = {
  buttonText: PropTypes.string.isRequired,
  setItems: PropTypes.func.isRequired,
  items: PropTypes.array.isRequired
};

export default Buttons
