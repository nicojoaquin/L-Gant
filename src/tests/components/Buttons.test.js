import React from 'react';
import {shallow} from 'enzyme';
import Buttons from '../../components/main/products/Buttons'


describe('Prueba en <Buttons />', () => {
  
  const buttonText = 'Agregar producto';
  const setItems = () => {
    "funcion"
  };
  const items = [];

  const wrapper = shallow(<Buttons buttonText = {buttonText} setItems = {setItems} items = {items} />);
  
  test('debe mostrar el componente correctamente', () => {
   
    expect( wrapper ).toMatchSnapshot();

  });

  test('debe tener un texto en el boton y la clase debe ser "addButton"', () => {
    
    const button = wrapper.find('.addButton');
    expect(button.text().trim()).toBe(buttonText);
    expect(button.prop('className').includes('addButton')).toBe(true);

  });
    

});


