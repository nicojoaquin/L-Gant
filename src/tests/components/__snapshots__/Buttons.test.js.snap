// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <Buttons /> debe mostrar el componente correctamente 1`] = `
<div
  className="buttons"
>
  <button
    className="addButton"
    disabled={false}
    onClick={[Function]}
  >
    Agregar producto
  </button>
  <button
    className="changeColor"
    onClick={[Function]}
  >
    Cambiar color
  </button>
</div>
`;
